fulp_modules/main_features/bloodsuckers/code/powers/feed.dm

```diff
@@ -194,133 +210,119 @@
+	ADD_TRAIT(user, TRAIT_IMMOBILIZED, BLOODSUCKER_TRAIT) // Prevents spilling blood accidentally.
+	. = ..()
 
-			if(amSilent)
-				to_chat(user, "<span class='warning'>Your feeding has been interrupted...but [target.p_they()] didn't seem to notice you.<span>")
-			else
-				to_chat(user, "<span class='warning'>Your feeding has been interrupted!</span>")
-				user.visible_message("<span class='danger'>[user] is ripped from [target]'s throat. [target.p_their(TRUE)] blood sprays everywhere!</span>", \
-						 			 "<span class='userdanger'>Your teeth are ripped from [target]'s throat. [target.p_their(TRUE)] blood sprays everywhere!</span>")
-				REMOVE_TRAIT(user, TRAIT_IMMOBILIZED, BLOODSUCKER_TRAIT)
-				// Deal Damage to Target (should have been more careful!)
-				if(iscarbon(target))
-					var/mob/living/carbon/C = target
-					C.bleed(15)
-				playsound(get_turf(target), 'sound/effects/splat.ogg', 40, 1)
-				if(ishuman(target))
-					var/mob/living/carbon/human/H = target
-					var/obj/item/bodypart/head_part = H.get_bodypart(BODY_ZONE_HEAD)
-					if(head_part)
-						head_part.generic_bleedstacks += 5
-				target.add_splatter_floor(get_turf(target))
-				user.add_mob_blood(target) // Put target's blood on us. The donor goes in the ( )
-				target.add_mob_blood(target)
-				target.take_overall_damage(10)
-				INVOKE_ASYNC(target, /mob.proc/emote, "scream")
+/datum/action/bloodsucker/feed/UsePower(mob/living/user)
+	var/mob/living/target = feed_target
+	var/datum/antagonist/bloodsucker/bloodsuckerdatum = IS_BLOODSUCKER(user)
+	var/datum/antagonist/vassal/vassaldatum = IS_VASSAL(user)
+//	if(!..()) // We're using our own checks below, becuase we have a TARGET to keep track of.
+//		return
 
-			return
+	// Did we deactivate this manually? Let's end it here.
+	if(!active)
+		return
+	if(!ContinueActive(user, target))
+		if(amSilent)
+			to_chat(user, "<span class='warning'>Your feeding has been interrupted...but [target.p_they()] didn't seem to notice you.<span>")
+		else
+			to_chat(user, "<span class='warning'>Your feeding has been interrupted!</span>")
+			user.visible_message("<span class='danger'>[user] is ripped from [target]'s throat. [target.p_their(TRUE)] blood sprays everywhere!</span>", \
+					 			 "<span class='userdanger'>Your teeth are ripped from [target]'s throat. [target.p_their(TRUE)] blood sprays everywhere!</span>")
+			// Deal Damage to Target (should have been more careful!)
+			if(iscarbon(target))
+				var/mob/living/carbon/C = target
+				C.bleed(15)
+			playsound(get_turf(target), 'sound/effects/splat.ogg', 40, 1)
+			if(ishuman(target))
+				var/mob/living/carbon/human/H = target
+				var/obj/item/bodypart/head_part = H.get_bodypart(BODY_ZONE_HEAD)
+				if(head_part)
+					head_part.generic_bleedstacks += 5
+			target.add_splatter_floor(get_turf(target))
+			user.add_mob_blood(target) // Put target's blood on us. The donor goes in the ( )
+			target.add_mob_blood(target)
+			target.take_overall_damage(10)
```

Comment:
this should be applied to the target's head if possible, since you were feeding on their neck.

Reply:
This is only called if it isnt silent, so only when aggressive.